src/CMakeLists.txt

```diff
@@ -20,17 +20,54 @@ foreach(src ${synchronization_sources})
 	list(APPEND argo_sources synchronization/${src})
 endforeach(src)
 
+
+# exactly one of these must be enabled. below is some code to ensure this.
+option(ARGO_VM_SHM
+	"Handle virtual addresses using POSIX shared memory. Size-limited." ON)
+option(ARGO_VM_ANONYMOUS
+	"Handle virtual addresses using anonymously-mapped memory. Slow." OFF)
+option(ARGO_VM_MEMFD
+	"Handle virtual addresses using an anonymous memory file. Requires kernel 3.17+." OFF)
+
+set(ARGO_VM_COUNT 0 CACHE INTERNAL "Check for multiple virtual address handlers")
+unset(vm_libs)
+
+if(ARGO_VM_SHM)
+	set(vm_sources shm.cpp)
```

Comment:
https://cmake.org/pipermail/cmake/2012-February/049286.html
that convention may exist, but I think we are not following it in numerous places.